### ISSUE TITLE
rake db:mongoid:create_indexes not working for rails engine components

### ISSUE THREAD

**Issue by agenteo** [mongoid/mongoid#3868]
Hi,

I am using a rails engine as a component of my rails application. When running the task `rake db:mongoid:create_indexes` from the main app it doesn't load the inner engine mongoid models.

Currently I am overriding the `db:mongoid:load_method` rake task and manually load the models I need.

Ideally I should be able to run: ` Rails::Mongoid.load_models(DomainLogic::Engine)` in this overridden load_method task.

But there is an issue in `lib/rails/mongoid.rb` `load_models method`. When running `app.config.paths['app/models'].each` it is returning a local path 'app/model'.

If instead you run `app.config.paths["app/models"].to_a.each do |path|` the returned path would be a full path and would allow to load mongoid files in engines.

What's your take on this? Am I missing some other workaround for this?

Basically I'd like to rely on mongoid index system for my engines models.

**Comment by BaylorRae** [comment]
You saved me a ton of time figuring out how to properly access model paths from the engine. Thank you

It's possible to redefine the `load_models` task to manually require the models from your engine. I simply copied the source from [Rails::Mongoid#load_models](https://github.com/mongoid/mongoid/blob/master/lib/rails/mongoid.rb) into my app.

```ruby
# in your application
# lib/tasks/load_models.rake
namespace :db do
  namespace :mongoid do

    # include models from engine
    task :load_models do
      YourEngineName::Engine.config.paths['app/models'].to_a.each do |path|
        preload = ::Mongoid.preload_models
        if preload.resizable?
          files = preload.map { |model| "#{path}/#{model}.rb" }
        else
          files = Dir.glob("#{path}/**/*.rb")
        end

        files.sort.each do |file|
          require_dependency(file)
        end
      end
    end

  end
end
```

**Comment by arthurnn** [comment]
Can you double check if this is still a problem on latest Mongoid? 
thanks

**Comment by BaylorRae** [comment]
I believe so. This is a quick test I ran in `rails console` to check.

```ruby
Loading development environment (Rails 4.1.10)
irb(main):001:0> Rails::Mongoid.load_models(AppModels::Engine)
=> ["app/models"]
irb(main):002:0> Mongoid.models
=> []
irb(main):003:0> AppModels::Engine.config.paths['app/models'].to_a.each do |path|
irb(main):004:1*   # truncated...
irb(main):014:1> end
=> ["/Users/baylorrae/Work/app/app_models/app/models"]
irb(main):015:0> Mongoid.models
=> [AppModels::Card, AppModels::DeviceToken, AppModels::User, AppModels::UserCard]
irb(main):016:0> Mongoid::VERSION
=> "4.0.2"
```

**Comment by krzysztofsciubisz** [comment]
I confirm having the same issue. Version `4.0.2`.

**Comment by nofxx** [comment]
Working fin in 5 beta. To be clear to some coming thru:

Create on your app (not the engine) a lib/tasks/load_model.rake
```
namespace :db do
  namespace :mongoid do

    # include models from engine
    task :load_models do
        Rails::Mongoid.load_models(SomeNice::Engine)
    end
  end
end
```
